utils/utils.go

```diff
@@ -0,0 +1,394 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pulumi/pulumi/pkg/apitype"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Each resource request will be fetched from the Kubernetes API Server at a
+// max of 12 retries, with 15 seconds in between each attempt.
+// This creates a max wait time of upto 3 minutes that a resource request
+// must successfully return within, before moving on.
+
+// MaxRetries is the maximum number of retries that a resource will be
+// attempted to be fetched from the Kubernetes API Server.
+const MaxRetries = 12
+
+// RetryInterval is the number of seconds to sleep in between requests
+// to the Kubernetes API Server.
+const RetryInterval = 15
+
+// RunEKSSmokeTest instantiates the EKS Smoke Test based on the Pulumi stack
+// resources and the exported kubeconfig outputs.
+func RunEKSSmokeTest(t *testing.T, resources []apitype.ResourceV3, kubeconfigs ...interface{}) {
+	// Map the cluster name to the KubeAccess client-go tool bag.
+	kubeAccess, err := MapClusterToKubeAccess(kubeconfigs...)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Map the cluster name to the total desired Node count across all
+	// NodeGroups.
+	clusterNodeCount, err := MapClusterToNodeCount(resources)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Run the smoke test against each cluster, expecting the total desired
+	// Node count.
+	for clusterName := range kubeAccess {
+		fmt.Printf("Testing Cluster: %s\n", clusterName)
+		t.Logf("Testing Cluster: %s\n", clusterName)
+		clientset := kubeAccess[clusterName].Clientset
+		EKSSmokeTest(t, clientset, clusterNodeCount[clusterName])
+	}
+}
+
+// EKSSmokeTest runs a checklist of operational successes required to deem the
+// EKS cluster as successfully running and ready for use.
+func EKSSmokeTest(t *testing.T, clientset *kubernetes.Clientset, desiredNodeCount int) {
+	// Print API Server info.
+	APIServerVersionInfo(t, clientset)
+
+	// Run all tests.
+	AssertEKSConfigMapReady(t, clientset)
+	AssertAllNodesReady(t, clientset, desiredNodeCount)
+	AssertAllPodsReady(t, clientset)
```

Comment:
Ah regrouped with @metral offline -- I was suggesting that these validations should go inside the `@pulumi/eks` package itself, vs. only in the tests. This would give the EKS library a sustained competitive advantage over alternatives, since ~no other systems allow this kind of post-validation.

Reply:
I'll plan to incorporate these new EKS validations within `pulumi/eks` for post-deployment validation in a future PR.

Reply:
Integration of the validations post-deployment is being tracked in https://github.com/pulumi/pulumi-eks/issues/124.